src/tools/pub/cmd/index.md

```diff
@@ -83,24 +104,40 @@ transformers before invoking the specified script.
 * The [`pub run`](/tools/pub/cmd/pub-run) command invokes a Dart script in your
   package, or in one of its dependencies.
 
-* The [`pub global`](/tools/pub/cmd/pub-global) command lets you work with globally
-  available packages that can be run when you are not currently inside
+* The [`pub global`](/tools/pub/cmd/pub-global) command lets you work with
+  globally available packages that can be run when you are not currently inside
   a package.
 
-## Deployment
+## Building and deploying apps and packages
+
+Pub supports building and deploying apps and packages. Some information
+is specific to web apps or command-line apps.
+
+### Sharing apps and packages
 
-When you are ready to deploy your app, the
-[`pub build`]({{site.webdev}}/tools/pub/pub-build) command
-creates the `build` directory, compiles the Dart code, and places
-the assets into the build directory.
+To share your Dart packages with the world, you can
+use the [`pub publish`](/tools/pub/cmd/pub-lish) command to upload the
+package to [pub.dartlang.org](https://pub.dartlang.org). The
+[`pub uploader`](/tools/pub/cmd/pub-uploader) command enables specific
+users to modify and upload new versions of your package.
 
-## Publication
+### Web apps
+
+Once you're ready to deploy a web app,
+use [`pub build`]({{site.webdev}}/tools/pub/pub-build)
+to generate the final files. This command creates the `build` directory,
+compiles the Dart code, and places all necessary files into the build
+directory.
+
+### Command-line apps
 
-If you want to share your Dart packages with the world, you can
-use the [`pub publish`](/tools/pub/cmd/pub-lish) command to upload your package to
-[pub.dartlang.org](https://pub.dartlang.org). The
-[`pub uploader`](/tools/pub/cmd/pub-uploader) command enables specific users
-to modify and upload new versions of your package.
+If you publish your scripts to pub.dartlang.org, or share them on
+GitHub, consider adding the
```

Comment:
You can do this even if you don't publish your packages. I use stagehand locally all the time when I'm prepping a new release. Maybe:

> For any package that contains scripts (anything under the bin/ directory), consider adding the executables tag to the pubspec file....